### ISSUE TITLE
[ENH] add test that checks all API reference links to be alive

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#3335]
We should test that all API reference links are alive.

Not sure how to do that, an example of unnoticed link breakage is https://github.com/alan-turing-institute/sktime/issues/3314 as highlighted by @juliencarponcy

**Comment by achieveordie** [comment]
I see a check for building the docs, but not a relevant job, why is that happening? Maybe I missed something. I'd look at it once again.

Anyways the solution that I can think of is to add another job that is dependent on the successful building of the docs. This job crawls through the API reference page and checks whether none of the links gives "Maze Found". 

Does that sound like a plausible solution?

**Comment by fkiraly** [comment]
> Does that sound like a plausible solution?

Yes, it does! If it is not too hard to implement?

**Comment by nikitashrivastava29** [comment]
Don't know if this would be relevant or not, but I was wondering if running tests on Postman to test all the routes would do the trick? (essentially automate the API testing)

**Comment by fkiraly** [comment]
@nikitashrivastava29, not sure what you propose - could you elaborate?
Help with this issue is appreciated!

**Comment by jgyasu** [comment]
Hmm, I see. One way is to use `beautifulsoup` to crawl through all the pages and get the links in them and then using `requests` the status code of all the links can be checked and based on the returned status code, the broken links can be identified.

Assign this to me @fkiraly. I will look into this.

**Comment by fkiraly** [comment]
Thanks! I hope this is not too much work - we would probably also only want to follow sktime internal links (pointing to sktime.net)?

**Comment by jgyasu** [comment]
> Thanks! I hope this is not too much work - we would probably also only want to follow sktime internal links (pointing to sktime.net)?

Yes, a regex filter checking for _sktime.net_ string in the links can do the job. But I wanted to know if I should create a separate file for this test under the `tests` directory or it should go in `_base.py` or `_meta.py`? Also, should the test be only confined to the links in the API reference docs or all the directories in the docs?

**Comment by jgyasu** [comment]
Update: Crawling the websites doesn't help because of how Sphinx generates links

**Comment by fkiraly** [comment]
> I wanted to know how to get all the documentation pages? Should I just check in https://www.sktime.net/en/stable/api_reference.html and all the pages linked inside it?

Hm, the full transition graph starting from sktime.net, within sktime.net, should give all pages, including `estimator_overview.html`, which can be accessed directly by clicking on the menu item on the top:
![image](https://github.com/user-attachments/assets/4ae13ee9-e3fc-4b1c-80cb-4552d90a02cd)
- why is this page not obtained in crawling?

**Comment by fkiraly** [comment]
> Also in which file and directory should this test go?

Good question - I would say:

* this should be a separate CI step, rather than a test
* how about the `build_tools` folder?
* the CI step would be added to `.github/workflows` and call that file
* the test logic itself can go in `tests`
